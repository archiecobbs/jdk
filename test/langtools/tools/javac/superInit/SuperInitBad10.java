/*
 * @test /nodynamiccopyright/
 * @bug 8194743
 * @summary Verify when present this()/super() must be invoked exactly once
 *
 * @compile/fail/ref=SuperInitBad10.out -XDrawDiagnostics SuperInitBad10.java
 */
class SuperInitBad10 {

    SuperInitBad10() {
        if (new Object().hashCode() > 0)
            super();
    }

    SuperInitBad10(int x) {
        for (int i = 0; i < 2; i++)
            super();
    }

    SuperInitBad10(char x) {
        switch (x) {
        case 'a':
            this((int)x);
            break;
        case 'b':
            super();
        default:
            break;
        }
    }
}
