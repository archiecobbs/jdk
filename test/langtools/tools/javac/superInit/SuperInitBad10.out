SuperInitBad10.java:13:5: compiler.err.superclass.constructor.might.not.have.been.invoked
SuperInitBad10.java:17:18: compiler.err.superclass.constructor.might.already.have.been.invoked
SuperInitBad10.java:18:5: compiler.err.superclass.constructor.might.not.have.been.invoked
SuperInitBad10.java:30:5: compiler.err.superclass.constructor.might.not.have.been.invoked
4 errors
